tests/agents/json-specs/sql_signature_examples.json

```
@@ -142,5 +142,9 @@
', '8B037581-08A7-481B-B8DB-5B28612A4951', '2F487CD3-B059-40AE-BA60-D2F88414FBDA', '35665D16-69A2-443F-A49E-FD5FBB0A1CA9', '95E52FB2-E931-4954-A491-CE110D21FAFE', 'e3b31408-331c-406d-9fed-ea6bea4a5424', '79d18a41-e4cb-453b-81a6-995959cff4e8', '5ba97bf9-7660-46a0-841d-2cc6d6494be1', 'B21722D6-630F-44DF-8578-EC1E29EDF7CC', '20A66E09-E48C-4A8A-B75B-67E153BD2F0C', 'F8758D03-E63B-428F-8F11-1A8A6F6A6516', 'F4BD0683-0C02-4C9A-B535-AE2C1CC82BFD', 'F3D40D8E-2A90-481D-A536-7393C30CD27F', 'E544A578-B74D-479F-B8DB-4020031517B5', 'E1AE1067-8448-47B1-8E60-CE77FE71AAA4', 'E18C4306-5E27-4CCE-969B-83D3487D5AD5', 'D35C7081-F013-4053-9F58-24C063C9DDCE', 'C88CDED9-4DC7-429D-8C4E-657E72CDA2E5', 'B4650761-EBD0-4D0B-801E-81FA27EC42F2', 'AD51D1FC-3C95-4215-8B3A-5D6209853512', 'AAFA9691-DC62-49CD-8B90-2848A9FABBEE', 'A40700DE-56ED-4BE3-B1DE-8976E81A9C75', '8F406559-8499-4143-A2F9-F2A080C4232F', '8A7CCB4E-25F3-40C9-8454-A3CBA26F31C7', '87B96787-2842-4C80-A8F9-E4BD9E37FFEF', '8433B6F3-4E96-4DD8-B31B-3327272B8F6B', '7C727602-B068-4E68-A460-9493F70E939C', '727C02C6-5D19-4382-ADDC-453C0CA56A9F', '60D99BF7-41C8-41FA-AE36-333F36827742', '5FB3D3DE-B35F-4CD3-8604-E95DB4782369', '5D0039B5-ABBC-454D-8BB1-10A330042FFA', '5C5FB635-8D90-4275-B321-DA360B556A18', '5484702B-9A4F-4214-AED8-6F244B409C38', '4AE1CECC-1646-40D1-954B-B0D8FF16A82D', '41E4B852-D3F4-4CFB-B5EF-B79CA25C1659', '3D1C8C5E-083C-4525-AB8D-8F7478181488', '3631280C-84EF-4891-A564-2263CFF78F18', '2F5C785B-EDFE-43DD-8DF1-9622FAD6BE3D', '24CF810D-9633-49F4-9F4B-F8AB8BD5ED03', '247EF179-6D68-45DF-BA60-4F63003FD216', '1D3C7E27-1C10-4E3D-95CD-86B9699E07EA', '1CBFA40D-3624-4B8B-851B-AFFEAB078E12', '161142B1-CC59-4DFA-B100-2316B0C62E80', '146D15A8-1CBD-43D3-8F4C-F568952F7A1F', '13316E84-8F98-48CA-86EA-5D01E9E61398', '0F460273-CCD7-4886-A1A4-1D5C96F97CC1', '0D749881-0F67-449E-97C5-1CF7F56915DA', '09A168BB-CF9E-4849-8FB9-088BFFBDD2C5', '079FE03E-AAC0-4627-9F7A-D83B91C6F675', '02F23B27-8596-4892-91AD-375F6F315273', 'FF2369C8-ABF1-4455-B0F3-9525BBE156AE', 'FCEBF5D4-BF2A-4679-8F66-44580CD97657', 'FC696FF3-CD0A-4AE2-9D42-E03B143A62C8', 'F20C53EB-4832-452D-8124-20B6E327D85E', 'EA1B9A54-8AEC-4C69-8F88-DCB15742A70F', 'E45F7274-52BE-4417-8F5F-BC59996AA583', 'ABA03842-4D52-4881-95F8-7C939CDDF916', '9D433B73-C4DC-4514-9C61-556D01C10173', '96C4F9BA-F264-4650-A520-251802E50807', '955924C6-6B9A-4435-B85B-03806104E161', '81D4931F-3643-471E-948E-E7763DB67AAA', '6E78B4D1-0462-40D5-8B8C-3C86D97B99DB', '6A2CFDAC-6742-4F0F-9378-432D6236D3D3', '66142156-73EF-4AC3-9E30-99D2F4ADD1F8', '64C7CD32-7688-491C-A063-9E83CB072B19', '62BE1BA9-2C4D-424F-B38C-0C15FA01B0DF', '541E54AA-76AF-4FE2-86CE-17A897DFBF88', '33B6716D-3318-43BA-A6B3-E4AA52A75185', '3014274C-C691-4D0F-BE1F-BC85747E8ADB', '0EC8716C-0E04-4AFA-82C6-94A908244A0B', 'CA1E2FC5-8809-4AF9-B05B-BF3D517BEF7A', 'B52EB6B2-3065-44B3-974C-CD5649211733', '9413088B-3954-422D-ACBA-B6B3744466D6', '807A9B76-5260-4EB7-A50D-F7E92E084628','8855e05b-7e38-4298-942c-afe5f4157e9e') AND ([s].[OnShelf] = CAST(1 AS bit))",
```

Suggestion that keeps the complexity of the query the same but makes it less verbose.
```suggestion
    "input": "SELECT [s].[Id], [s].[CreateTime] FROM [Speech] AS [s] WHERE [s].[Id] IN ('FCD84661-F36C-4DB1-8B92-BE0668027163', '14274997-dcc7-4a7a-b866-41a007b36b9b') AND ([s].[OnShelf] = CAST(1 AS bit))",
```

---

I was considering that but did not want to change the original test input data (no strong opinion though).

I don't think the actual length of the statement is something we wanted to test here, right @gregkalapos?

---

> I don't think the actual length of the statement is something we wanted to test here, right @gregkalapos?

Well, that's not the case :) The length of it was relevant.

That test data came out from this issue: https://github.com/elastic/apm-agent-dotnet/issues/1763, which was a performance issue in `Scanner` reported by users.

There was no issue with the parsing logic itself - so making it short or long won't make a difference, the test coverage is fine both ways. 

On the other hand we have a benchmark (added in https://github.com/elastic/apm-agent-dotnet/pull/1782/files) that indeed needs a long SQL statement to hit the issue reported in  https://github.com/elastic/apm-agent-dotnet/issues/1763.

So overall the point of this long statement was to have that as input for the benchmark - we'd loose that if we'd only have the short one.

So, what I'd suggest from the .NET Agent's point of view:
- Either we add the long version here and then we can use this in our benchmark as well
- Adding the short one won't make a huge difference (I don't think there is anything in the query itself which isn't covered already), so if we decide to do that, then I'd still prefer to keep the long one in .NET for the benchmark.

---

OK, that's useful feedback. Thanks @gregkalapos!
I was not aware of the fact that we use the same input data (`sql_signature_examples.json`) in the benchmark test as well.
I was only working with this data in the scope of `TestDbSignatures`.

Given this new information, I think I should withdraw this PR as it does not add additional value.
We can then continue the discussion in the `apm-agent-dotnet` repo. WDYT?

---

Sounds ok to me.